packaging_automation/prepare_release.py

```diff
@@ -0,0 +1,395 @@
+                                       head=upcoming_version_branch, body="")
+    print(f"### OK Pull request created. PR Number:{pr_result.number} PR URL: {pr_result.url}### ")
+
+
+def push_branch(upcoming_version_branch):
+    print(f"Pushing changes for {upcoming_version_branch} into remote origin...### ")
+    run(f"git push --set-upstream origin {upcoming_version_branch}")
+    print(f"### OK Changes pushed for {upcoming_version_branch}### ")
+
+
+def commit_changes_for_version_bump(project_name, project_version):
+    current_branch = get_current_branch()
+    print(f"### Committing changes for branch {current_branch}...### ")
+
+    run(f' git commit -a -m "Bump {project_name} version to {project_version} "')
+    print(f"### OK Changes committed for {current_branch}### ")
+
+
+def update_version_with_upcoming_version_in_citus_control(citus_control_file_path, upcoming_minor_version):
+    print(f"### Updating {citus_control_file_path} file with the upcoming version {upcoming_minor_version}...### ")
+    if not replace_line_in_file(citus_control_file_path, CITUS_CONTROL_SEARCH_PATTERN,
+                                f"default_version = '{upcoming_minor_version}-1'"):
+        raise ValueError(f"{citus_control_file_path} does not have match for version")
+    print(f"### OK {citus_control_file_path} file is updated with the upcoming version {upcoming_minor_version}...### ")
+
+
+def update_schema_version_with_upcoming_version_in_multi_extension_file(current_schema_version,
+                                                                        multi_extension_sql_path,
+                                                                        upcoming_minor_version, upcoming_version):
+    print(
+        f"### Updating schema version {current_schema_version} on {multi_extension_sql_path} "
+        f"file with the upcoming version {upcoming_version}...### ")
+    # TODO Append instead of replace may require
+    if not replace_line_in_file(multi_extension_sql_path,
+                                f"ALTER EXTENSION citus UPDATE TO '{current_schema_version}'",
+                                f"ALTER EXTENSION citus UPDATE TO '{upcoming_minor_version}-1';"):
+        raise ValueError(f"{multi_extension_sql_path} does not have match for version")
+    print(f"### OK Current schema version updated on {multi_extension_sql_path} to {upcoming_minor_version}### ")
+
+
+def get_current_schema_from_citus_control(citus_control_file_path: str) -> str:
+    print(f"### Reading current schema version from {citus_control_file_path}...### ")
+    current_schema_version = ""
+    with open(citus_control_file_path, "r") as cc_reader:
+        cc_file_content = cc_reader.read()
+        cc_line = find_nth_matching_line(cc_file_content, CITUS_CONTROL_SEARCH_PATTERN, 1)
+        schema_not_found = False
+        if len(cc_line) > 0:
+            line_parts = cc_line.split("=")
+            if len(line_parts) > 0:
```

Comment:
let's be more explicit:

```suggestion
            if len(line_parts) == 2:
```

Reply:
Fixed